### PR TITLE
HWKAPM-564 Added capability to disassociate a fragment builder from t…

### PR DESCRIPTION
**Pull request by objectiser**
…he thread, if an actively managed thread performs a particular activity

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Hi Pavol, could you review this one please? It should provide a fix to part of the problem found by Joel on vertx.

File: client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java [hunk 1]
```diff
@@ -1494,6 +1494,22 @@ public boolean isActive() {
...
     /* (non-Javadoc)
+     * @see org.hawkular.apm.client.api.SessionManager#deactivate()
+     */
+    @Override
+    public void deactivate() {
+        try {
+            if (fragmentManager.hasFragmentBuilder()) {
+                fragmentManager.clear();
+            }
+        } catch (Throwable t) {
```
Comment (pavolloffay) [inline client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java]: Is there more concrete exception?

Comment (objectiser) [inline client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java]: No. This is truly a "catch everything and don't dare impact the application" catch block.

**Comment by jotak** [comment]
This is good to me. I've tested it on my use case (which was, basically, request on "/" => produces something-I-don't-know bad, then any further request on a different path would fail). Now the request on "/" still fails for some reason (ie. nothing is sent to the APM server) but the subsequent requests on another path are fine.

**Comment by objectiser** [comment]
Thanks @pavolloffay and @jotak .

**Pull request merged**